webrtc/RTCIceTransport.html

```diff
@@ -114,81 +65,128 @@
 
-      assert_equals(candidatePair1.local.candidate, candidatePair2.remote.candidate,
-        'Expect selected local candidate of one pc is the selected remote candidate or another');
+    const [track] = await getTrackFromUserMedia('audio');
+    pc.addTransceiver(track);
 
-      assert_equals(candidatePair1.remote.candidate, candidatePair2.local.candidate,
-        'Expect selected local candidate of one pc is the selected remote candidate or another');
+    // setLocalDescription should initialize sender/receiver.transport
+    const offer = await pc.createOffer();
+    await pc.setLocalDescription(offer);
 
-      assert_equals(iceTransport1.role, 'controlling',
-        `Expect offerer's iceTransport to take the controlling role`);
+    const iceTransports =[
+      ...pc.getSenders(),
+      ...pc.getReceivers()
+    ]
+    .map(senderReceiver => senderReceiver.transport)
+    .filter(dtlsTransport => dtlsTransport) // filter null/undefined
+    .map(dtlsTransport => dtlsTransport.iceTransport);
 
-      assert_equals(iceTransport2.role, 'controlled',
-        `Expect answerer's iceTransport to take the controlled role`);
-    });
-  }, 'Two connected iceTransports should has matching local/remote candidates returned');
+    assert_greater_than_equal(iceTransports.length, 1, 'expect to find at least 1 RTCIceTransports');
 
-  promise_test(t => {
+    for (const iceTransport of iceTransports) {
+      validateUnconnectedIceTransport(iceTransport);
+    }
+  }, 'RTP - Unconnected iceTransport should have empty remote candidates and selected pair');
+
+  promise_test(async t => {
     const pc1 = new RTCPeerConnection();
     t.add_cleanup(() => pc1.close());
+
     const pc2 = new RTCPeerConnection();
     t.add_cleanup(() => pc2.close());
+
     pc1.createDataChannel('');
 
-    // setRemoteDescription(answer) without the other peer
-    // setting answer it's localDescription
-    return pc1.createOffer()
-    .then(offer =>
-      pc1.setLocalDescription(offer)
-      .then(() => pc2.setRemoteDescription(offer))
-      .then(() => pc2.createAnswer()))
-    .then(answer => pc1.setRemoteDescription(answer))
-    .then(() => {
-      const iceTransport = getIceTransportFromSctp(pc1);
-
-      assert_array_equals(iceTransport.getRemoteCandidates(), [],
-        'Expect iceTransport to not have any remote candidate');
-
-      assert_equals(iceTransport.getSelectedCandidatePair(), null,
-        'Expect selectedCandidatePair to be null');
-    });
-  }, 'Unconnected iceTransport should have empty remote candidates and selected pair');
+    // pc1.setRemoteDescription(answer) without pc2 setting answer to it's
+    // localDescription. This is because pc1.sctp is only set after setting
+    // answer description.
+    const offer = await pc1.createOffer()
+
+    await pc1.setLocalDescription(offer)
+    await pc2.setRemoteDescription(offer)
+    const answer = await pc2.createAnswer()
+    await pc1.setRemoteDescription(answer)
```

Comment:
Tip: could be inlined, for fewer names polluting the cognitive stack space:
```js
await pc1.setRemoteDescription(await pc2.createAnswer());
```
Also, missing `;` semicolons at EOLs here.